src/components/DuffelNGSView/NGSTable.tsx

```diff
@@ -0,0 +1,281 @@
+} from "./lib/group-offers-for-ngs-view";
+import { SortDirection, sortNGSRows } from "./lib/sort-ngs-rows";
+import { NGSSliceFareCard } from "./NGSSliceFareCard";
+import { NGSShelfInfoCard } from "./NGSShelfInfoCard";
+import { SliceSummary } from "./SliceSummary";
+import { OfferSliceModal } from "@components/OfferSliceModal/OfferSliceModal";
+
+export interface NGSTableProps {
+  offers: OfferWithNGS[];
+  sliceIndex: number;
+  previousSliceKeys: string[]; // For filtering the current set of offers
+  onSelect: (offerId: string, sliceKey: string) => void;
+}
+
+type OfferPosition = {
+  row: number;
+  shelf: NGSShelf;
+};
+
+function getPreviousShelf(shelf: NGSShelf): NGSShelf | null {
+  const previousShelf = +shelf - 1;
+  return previousShelf === 0 ? null : (previousShelf.toString() as NGSShelf);
+}
+
+function getNextShelf(shelf: NGSShelf): NGSShelf | null {
+  const nextShelf = +shelf + 1;
+  return nextShelf > 5 ? null : (nextShelf.toString() as NGSShelf);
+}
+
+const getPreviousOffer = (
+  rows: NGSOfferRow[],
+  expandedOffer: OfferPosition,
+): OfferWithNGS | null => {
+  const previousShelf = getPreviousShelf(expandedOffer.shelf);
+  if (!previousShelf) {
+    return null;
+  }
+  const previousOffer = rows[expandedOffer.row][previousShelf];
+  if (previousOffer) {
+    return previousOffer;
+  }
+  return getPreviousOffer(rows, {
+    row: expandedOffer.row,
+    shelf: previousShelf,
+  });
+};
+
+const getNextOffer = (
+  rows: NGSOfferRow[],
+  expandedOffer: OfferPosition,
+): OfferWithNGS | null => {
+  const nextShelf = getNextShelf(expandedOffer.shelf);
+  if (!nextShelf) {
+    return null;
+  }
+  const nextOffer = rows[expandedOffer.row][nextShelf];
+  if (nextOffer) {
+    return nextOffer;
+  }
+  return getNextOffer(rows, { row: expandedOffer.row, shelf: nextShelf });
+};
+
+export const NGSTable: React.FC<NGSTableProps> = ({
+  offers,
+  sliceIndex,
+  onSelect,
+  previousSliceKeys,
+}) => {
+  const [selectedColumn, setSelectedColumn] = React.useState<NGSShelf | null>(
+    null,
+  );
+  const [sortShelf, setSortShelf] = React.useState<NGSShelf | null>(null);
+  const [sortDirection, setSortDirection] =
+    React.useState<SortDirection>("asc");
+  const [rows, setRows] = React.useState<NGSOfferRow[]>(
+    groupOffersForNGSView(offers, sliceIndex, previousSliceKeys),
+  );
+  const [expandedOffer, setExpandedOffer] =
+    React.useState<OfferPosition | null>(null);
+  const [isOfferSliceModalOpen, setIsOfferSliceModalOpen] =
+    React.useState<boolean>(false);
+
+  React.useEffect(() => {
+    setSelectedColumn(null);
+    setSortShelf(null);
+    setExpandedOffer(null);
+    setIsOfferSliceModalOpen(false);
+    setRows(groupOffersForNGSView(offers, sliceIndex, previousSliceKeys));
+  }, [previousSliceKeys]);
+
+  React.useEffect(() => {
+    if (sortShelf) {
+      const sortedRows = sortNGSRows(rows, sortShelf, sortDirection);
+      setRows(sortedRows);
+    }
+  }, [sortShelf, sortDirection]);
```

Comment:
I don't think we need this? Sorted rows is the output of some state, but itself doesn't need to be persisted, right? 
Something like this I guess?
```suggestion
      const rows = groupOffersForNGSView(offers, sliceIndex, previousSliceKeys)
      const sortedRows = sortShelf ? rows : sortNGSRows(rows, sortShelf, sortDirection); 
```
And then use sortedRows below?

Reply:
Yeah you're right, not sure why I added that in the first place :/